src/AbstractAttachment.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * This file is part of the CalendArt package
+ *
+ * For the full copyright and license information, please view the LICENSE file
+ * that was distributed with this source code.
+ *
+ * @copyright Wisembly
+ * @license   http://www.opensource.org/licenses/MIT-License MIT License
+ */
+
+namespace CalendArt;
+
+use Datetime;
+use InvalidArgumentException;
+
+/**
+ * Represents an Attachment in a Event
+ *
+ * Like all generic objects, this object should be extended by the adapter
+ */
+abstract class AbstractAttachment
+{
+    /** @var string name of the attachment */
+    protected $name;
+
+    /** @var string URI of this attachment */
+    protected $uri;
+
+    public function __construct($name, $uri)
+    {
+        $this->name = $name;
+        $this->uri = $uri;
+    }
+
+    /** @return mixed */
+    abstract public function getId();
+
+    /** @return string */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /** @return $this */
+    public function setName($name)
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /** @return string */
+    public function getUri()
+    {
+        return $this->uri;
+    }
+
+    /** @return $this */
+    public function setUri($uri)
+    {
+        $this->uri = $uri;
+
+        return $this;
+    }
```

Comment:
IMO we shoud have `getContents` (and no `SetContents`... you do not change the content or the uri for any reason). IT's up to the adapter to be able to return this stuff.